File: Лабораторная работа 5/AutoTestJava/AutoTest/src/test/java/ru/auto/HomePage.java
package ru.auto;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.Keys;

public class HomePage extends BasePage {
    private static final String URL = "https://auto.ru/";
    private static final String HISTORY_URL = "https://auto.ru/history/";
    private static final String CATALOG_URL = "https://auto.ru/catalog/cars/";

    // Locators
    private final By inputSearch = By.xpath("//div[contains(@class, 'SearchLineSuggest__input')]//input");
    private final By searchResults = By.xpath("//div[contains(@class,'ListingHead__content')]//h1");
    private final By emptySearch = By.xpath("//div[contains(@class, 'Index')]//div");
    private final By filterPriceFrom = By.xpath("//label[contains(@class, 'TextInput TextInput_size_l TextInput_has-clear TextInput_has-placeholder TextInput_placeholderShowWithValue')]//input");
    private final By filterPriceTo = By.xpath("//label[contains(@class, 'TextInput TextInput_size_l TextInput_has-clear TextInput_has-placeholder TextInput_placeholderShowWithValue')]//input");
    private final By filterButton = By.xpath("//button[contains(@class,'Button2')]");

    private final By catalogSection = By.xpath("//div[contains(@class, 'Index')]");
    private final By catalogFirstPage = By.xpath("//div[contains(@class,'BaseApp__content')]");

    private final By vinInput = By.xpath("//div[contains(@class, 'VinCheckInput3__controls-EZidM')]//input");
    private final By vinCheckButton = By.xpath("//button[contains(@class,'Button2')]");

    public HomePage(WebDriver driver) {
        super(driver);
    }

    // Open URL
    public void open() {
        openUrl(URL);
    }

    // Go to History
    public void goToHistory() {
        openUrl(HISTORY_URL);
    }

    // Go to Catalog
    public void goToCatalog() {
        openUrl(CATALOG_URL);
    }

    // Execute Search
    public void executeSearch(String searchText) {
        click(inputSearch, 10);
        type(inputSearch, searchText, 10);
        find(inputSearch).sendKeys(Keys.RETURN);
    }

    // Apply Price Filter
    public void applyPriceFilter(int minPrice, int maxPrice) {
        type(filterPriceFrom, String.valueOf(minPrice), 10);
        type(filterPriceTo, String.valueOf(maxPrice), 10);
        click(filterButton, 10);

        // Задержка перед закрытием браузера (чтобы увидеть результат)
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    // Open Catalog
    public void openCatalog() {
        click(catalogSection, 10);
    }

    // Check VIN
    public void checkVin(String vin) {
        type(vinInput, vin, 5);
        click(vinCheckButton, 5);

        // Задержка перед закрытием браузера (чтобы увидеть результат)
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    // Get First Search Result
    public String getFirstSearchResult() {
        return getText(searchResults, 10);
    }

    // Get Empty Search Result
    public String getEmptySearch() {
        return getText(emptySearch, 10);
    }

    // Get First Catalog Result
    public String getFirstCatalogResult() {
        return getText(catalogFirstPage, 10);
    }
}

File: AutoTestSharp/AutoTest/AutoTest/HomePage.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AutoTest
{
    public class HomePage : BasePage
    {
        private const string URL = "https://auto.ru/";
        private const string HISTORY_URL = "https://auto.ru/history/";
        private const string CATALOG_URL = "https://auto.ru/catalog/cars/";

        // Locators
        private readonly By inputSearch = By.XPath("//div[contains(@class, 'SearchLineSuggest__input')]//input");
        private readonly By searchResults = By.XPath("//div[contains(@class,'ListingHead__content')]//h1");
        private readonly By emptySearch = By.XPath("//div[contains(@class, 'Index')]//div");
        private readonly By filterPriceFrom = By.XPath("//label[contains(@class, 'TextInput TextInput_size_l TextInput_has-clear TextInput_has-placeholder TextInput_placeholderShowWithValue')]//input");
        private readonly By filterPriceTo = By.XPath("//label[contains(@class, 'TextInput TextInput_size_l TextInput_has-clear TextInput_has-placeholder TextInput_placeholderShowWithValue')]//input");
        private readonly By filterButton = By.XPath("//button[contains(@class,'Button2')]");
        private readonly By catalogSection = By.XPath("//div[contains(@class, 'Index')]");
        private readonly By catalogFirstPage = By.XPath("//div[contains(@class,'BaseApp__content')]");
        private readonly By vinInput = By.XPath("//div[contains(@class, 'VinCheckInput3__controls-EZidM')]//input");
        private readonly By vinCheckButton = By.XPath("//button[contains(@class,'Button2')]");

        public HomePage(IWebDriver driver) : base(driver) { }

        public void Open()
        {
            OpenUrl(URL);
        }

        public void GoToHistory()
        {
            OpenUrl(HISTORY_URL);
        }

        public void GoToCatalog()
        {
            OpenUrl(CATALOG_URL);
        }

        public void ExecuteSearch(string searchText)
        {
            Click(inputSearch, 10);
            Type(inputSearch, searchText, 10);
            Find(inputSearch).SendKeys(Keys.Enter);
        }

        public void ApplyPriceFilter(int minPrice, int maxPrice)
        {
            Type(filterPriceFrom, minPrice.ToString(), 10);
            Type(filterPriceTo, maxPrice.ToString(), 10);
            Click(filterButton, 10);
            Thread.Sleep(5000);  // Ждем результата фильтрации
        }

        public void OpenCatalog()
        {
            Click(catalogSection, 10);
        }

        public void CheckVin(string vin)
        {
            Type(vinInput, vin, 5);
            Click(vinCheckButton, 5);
            Thread.Sleep(5000);
        }

        public string GetFirstSearchResult()
        {
            return GetText(searchResults, 10);
        }

        public string GetEmptySearch()
        {
            return GetText(emptySearch, 10);
        }

        public string GetFirstCatalogResult()
        {
            return GetText(catalogFirstPage, 10);
        }
    }
}
